include/device.h

```diff
@@ -296,6 +296,20 @@ struct device_pm {
  * before they are accessed.
  */
 struct device_state {
+	/** Non-negative result of initializing the device.
+	 *
+	 * The absolute value returned when the device initialization
+	 * function was invoked, or `UINT8_MAX` if the value exceeds
+	 * an 8-bit integer.  If initialized is also set, a zero value
+	 * indicates initialization succeeded.
+	 */
+	unsigned int init_res : 8;
```

Comment:
I have the impression that initialization functions in general return a negative value in case of error, and we will end up always with `0` when it succeed or `255` on error  ... That means that use one bit would give the same info, We should probably keep it signed.

Reply:
nevermind, you invert the signal later